test/sequential/test-repl-persistent-history.js

```diff
@@ -66,17 +77,24 @@ const homedirErr = '\nError: Could not get the home directory.\n' +
                    'REPL session history will not be persisted.\n';
 const replFailedRead = '\nError: Could not open history file.\n' +
                        'REPL session history will not be persisted.\n';
-// File paths
-const fixtures = path.join(common.testDir, 'fixtures');
-const historyFixturePath = path.join(fixtures, '.node_repl_history');
-const historyPath = path.join(common.tmpDir, '.fixture_copy_repl_history');
-const historyPathFail = path.join(common.tmpDir, '.node_repl\u0000_history');
-const oldHistoryPath = path.join(fixtures, 'old-repl-history-file.json');
-const enoentHistoryPath = path.join(fixtures, 'enoent-repl-history-file.json');
-const defaultHistoryPath = path.join(common.tmpDir, '.node_repl_history');
-
+const sameHistoryFilePaths = '\nThe old repl history file has the same name ' +
+                             'and location as the new one i.e., ' +
+                             path.join(fixtures, '.empty-repl-history-file') +
+                             ' and is empty.\nUsing it as is.\n';
 
 const tests = [{
+  env: { NODE_REPL_HISTORY: emptyHistoryPath,
+         NODE_REPL_HISTORY_FILE: emptyHistoryPath },
+  test: [UP],
+  expected: [prompt, sameHistoryFilePaths, prompt]
+},
+{
+  env: { NODE_REPL_HISTORY: defaultHistoryPath,
```

Comment:
No, it works either way. I have refactored the code to let `os.homedir()` do it's thing :)